docs/flow/modin/engines/base/io.md

```diff
@@ -0,0 +1,33 @@
+## High-Level Data Import Operation Workflow
```

Comment:
Since this is developer documentation, I think it would be fine to have it as `modin/engines/base/io/README.md`. Does that make sense or do you think we should have a `developer` section of the documentation and put this on the readthedocs site?

Reply:
This documentation was added to the `docs/flow` directory as it was mentioned in the [epic header](https://github.com/modin-project/modin/issues/2934#issue-843252517). I think the decision where to place this docs should be made according to high-level plan of Modin documenting. @vnlitvinov what do you think?